x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/TextFormat.java

```diff
@@ -48,43 +46,21 @@
      */
     PLAIN_TEXT() {
         @Override
-        String format(RestRequest request, SqlQueryResponse response) {
-            BasicFormatter formatter = null;
-            Cursor cursor = null;
-            ZoneId zoneId = null;
-
-            // check if the cursor is already wrapped first
-            if (response.hasCursor()) {
-                Tuple<Cursor, ZoneId> tuple = Cursors.decodeFromStringWithZone(response.cursor());
-                cursor = tuple.v1();
-                zoneId = tuple.v2();
-                if (cursor instanceof TextFormatterCursor) {
-                    formatter = ((TextFormatterCursor) cursor).getFormatter();
-                }
-            }
-
-            // if there are headers available, it means it's the first request
-            // so initialize the underlying formatter and wrap it in the cursor
-            if (response.columns() != null) {
-                formatter = new BasicFormatter(response.columns(), response.rows(), TEXT);
-                // if there's a cursor, wrap the formatter in it
-                if (cursor != null) {
-                    response.cursor(Cursors.encodeToString(new TextFormatterCursor(cursor, formatter), zoneId));
-                }
-                // format with header
-                return formatter.formatWithHeader(response.columns(), response.rows());
-            } else if (formatter != null) { // should be initialized (wrapped by the cursor)
-                // format without header
-                return formatter.formatWithoutHeader(response.rows());
-            } else if (response.hasId()) {
-                // an async request has no results yet
-                return StringUtils.EMPTY;
```

Comment:
Why hasn't this one been kept?

Reply:
I'll try to revert the original logic to avoid regressions in the async API.